yt/visualization/particle_plots.py

```diff
@@ -243,15 +349,26 @@ def __init__(
         data_source=None,
         deposition="ngp",
         density=False,
+        *,
+        north_vector=None,
+        axis=None,
     ):
+        if axis is not None:
+            issue_deprecation_warning(
+                "The 'axis' argument is a deprecated alias for the 'normal' argument. ",
+                "4.2.0",
+                "4.3.0",
+            )
+            normal = axis
+        if normal is None:
+            raise ValueError(
+                "You must specify an axis name or normal vector for this plot."
+            )
```

Comment:
Matching the error message produced with, for instance `def foo(bar): pass; foo()`
```suggestion
            raise TypeError(
                "ParticleProjectionPlot() missing 1 required positional argument: 'normal'"
            )
```